### PR TITLE
Add integration tests

### PR DESCRIPTION
**Pull request by pbruski**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -13,16 +14,19 @@ configurations.all {
...
+                "org.seleniumhq.selenium" -> useVersion(seleniumVersion)
+                "org.slf4j" -> useVersion("1.7.25")
             }
         }
     }
 }
 
 dependencies {
-    api("com.atlassian.performance.tools:jira-actions:[2.1.0,4.0.0)")
+    api("com.atlassian.performance.tools:jira-actions:[3.10.2,4.0.0)")
```
Comment (pbruski) [inline build.gradle.kts]: Needed because of stability, not APIs.

**Comment by pbruski** [comment]
Travis is unable to run this build, I think it requires slightly more heap than jira-actinos. Is there a way around it? or should I just comment out the tests so that they can  only be run manually?

File: src/test/kotlin/com/atlassian/performance/tools/jirasoftwareactions/api/scenario/JiraSoftwareScenarioIT.kt [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                    }
+                }
+            }
+
+        val results = metrics.map { metric ->
+            metric.result
+        }
+        Assertions.assertThat(results).containsOnly(ActionResult.OK)
+        val viewIssueMetrics = metrics.filter {
+            VIEW_ISSUE.label.equals(it.label)
```
Comment (dagguh) [inline src/test/kotlin/com/atlassian/performance/tools/jirasoftwareactions/api/scenario/JiraSoftwareScenarioIT.kt]: Perhaps we could assert on one of the JSW actions?

File: build.gradle.kts (hunk 1 above)
Comment (dagguh) [reply build.gradle.kts]: You can leave it as is: `[2.1.0,4.0.0)`.
To update the locks, you can update only the one you need, plus the resulting conflicts:
```
./gradlew :dep --update-locks "com.atlassian.performance.tools:jira-actions,com.squareup.okio:okio,com.squareup.okhttp3:okhttp,com.google.guava:guava,net.bytebuddy:byte-buddy"
```
This way, the POM from `./gradlew genPom` is left intact:
```xml
    <dependency>
      <groupId>com.atlassian.performance.tools</groupId>
      <artifactId>jira-actions</artifactId>
      <version>[2.1.0,4.0.0)</version>
      <scope>compile</scope>
    </dependency>
```

- [ ] don't break major POM compatibility

File: build.gradle.kts [hunk 3]
```diff
@@ -46,9 +51,24 @@ fun log4j(
...
+val testIntegration = task<Test>("testIntegration") {
+    testLogging.exceptionFormat =  org.gradle.api.tasks.testing.logging.TestExceptionFormat.FULL
+    testLogging.showStackTraces = true
+    testLogging.showExceptions = true
+    testLogging.showCauses = true
+    maxHeapSize = "2G"
+    include("**/*IT.class")
+}
+
+tasks["check"].dependsOn(testIntegration)
```
Comment (dagguh) [inline build.gradle.kts]: `testIntegration` works locally for me:
```
BUILD SUCCESSFUL in 13m 41s
3 actionable tasks: 3 executed
```

If it doesn't work on CI, then let's remove this `check -> testIntegration` dependency

File: src/main/kotlin/com/atlassian/performance/tools/jirasoftwareactions/api/page/BrowseBoardsPage.kt [hunk 4]
```diff
@@ -28,20 +28,27 @@ class BrowseBoardsPage(
...
+            elementToBeClickable(By.cssSelector("#ghx-manage-boards-filter a"))
+        ).click()
+
+        driver.wait(
+            Duration.ofSeconds(5),
+            elementToBeClickable(By.className("type-filter-scrum"))
+        ).click()
+        
+        driver.wait(
+            Duration.ofSeconds(60),
```
Comment (dagguh) [inline src/main/kotlin/com/atlassian/performance/tools/jirasoftwareactions/api/page/BrowseBoardsPage.kt]: It's too long of a timeout for production. `BrowseBoardsPage` is widely used in the wild.

File: src/main/kotlin/com/atlassian/performance/tools/jirasoftwareactions/api/scenario/JiraSoftwareScenario.kt [hunk 5]
```diff
@@ -1,4 +1,4 @@
-package com.atlassian.performance.tools.jirasoftwareactions.api
+package com.atlassian.performance.tools.jirasoftwareactions.api.scenario
```
Comment (dagguh) [inline src/main/kotlin/com/atlassian/performance/tools/jirasoftwareactions/api/scenario/JiraSoftwareScenario.kt]: This effectively deletes the `com.atlassian.performance.tools.jirasoftwareactions.api.JiraSoftwareScenario` class, which breaks source and binary compat.

- [ ] don't break JSS

**Comment by dagguh** [comment]
WRT red build, it reads:
```
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.

Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received
```

The workaround is: https://github.com/atlassian/jira-actions/commit/1efff0e12387340fba81899f88ebdfbb910d5865

Bonus workaround for Travis lack of test reports are Gradle build scans: https://github.com/atlassian/jira-actions/commit/5ab90e6a7012b99d56d65cb8b791d25de4dfac35

If after all that it fails with `ContainerLaunchException`, then [my previous attempts have met the same fate](https://github.com/atlassian/jira-actions/pull/29#commits-pushed-9fd2b24). OTOH I was able to [reproduce the problem in the docker-infra itself](https://github.com/atlassian/docker-infrastructure/pull/5).

**Comment by dagguh** [comment]
Here's an alternative way: https://github.com/atlassian/virtual-users/pull/7
It says, screw `docker-infrastructure`, let me just get SSH Ubuntu from Docker and `infrastructure` will do the rest.

File: src/main/kotlin/com/atlassian/performance/tools/jirasoftwareactions/api/page/BrowseBoardsPage.kt (hunk 4 above)
Comment (pbruski) [reply src/main/kotlin/com/atlassian/performance/tools/jirasoftwareactions/api/page/BrowseBoardsPage.kt]: The reason why it's 60 here is that it should not be needed, unless it's a chromedriver problem which means it needs to be fixed in chromedriver or the condition needs to be different.

**Comment by pbruski** [comment]
Trying Travis patience...

**Pull request merged**